Update the link to the Cairo contract code in address section (#911)
starknet-io/starknet-docs: components/Starknet/modules/architecture_and_concepts/pages/Smart_Contracts/contract-address.adoc

It also thwarts replay attacks by influencing the transaction hash with a unique sender address.
====

> You can find more detail of the address computation in the https://github.com/starkware-libs/cairo/blob/2c96b181a6debe9a564b51dbeaaf48fa75808d53/corelib/src/starknet/contract_address.cairo[Cairo code repository].

Reviewer: ```suggestion
For more information on the address computation, see the https://github.com/starkware-libs/cairo/blob/2c96b181a6debe9a564b51dbeaaf48fa75808d53/corelib/src/starknet/contract_address.cairo[Cairo code repository].
```
